packages/cosmic-swingset/src/launch-chain.js

```diff
@@ -522,8 +530,11 @@ export async function launch({
     // First, push all newActions onto the end of the inboundQueue,
     // remembering that inboundQueue might still have work from the
     // previous block
-    for (const a of newActions) {
-      inboundQueue.push(a);
+    let actionNum = 0;
+    for (const action of newActions) {
+      const inboundNum = `${blockHeight}-${actionNum}`;
+      inboundQueue.push({ action, inboundNum });
+      actionNum += 1;
       inboundQueueMetrics.incStat();
```

Comment:
I plan on adding a new slog event here to root the causality graph in the inbound queue acceptance.